rpc/client/mock/abci.go

```diff
@@ -0,0 +1,186 @@
+package mock
+
+import (
+	abci "github.com/tendermint/abci/types"
+	"github.com/tendermint/tendermint/rpc/client"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// ABCIApp will send all abci related request to the named app,
+// so you can test app behavior from a client without needing
+// an entire tendermint node
+type ABCIApp struct {
+	App abci.Application
+}
+
+func (a ABCIApp) _assertABCIClient() client.ABCIClient {
```

Comment:
we don't need this anymore. It was only an issue because there was an embedded interface type in abci.Application (the BlockchainAware bit). Now that has been moved into the abci.Application, so as long as that interface is being passed around (it is, everywhere), then the compiler already type checks